qt_dotgraph/src/qt_dotgraph/dot_to_qt.py

```diff
@@ -257,25 +277,36 @@ def dotcode_to_qt_items(self, dotcode, highlight_level, same_label_siblings=Fals
                 # hack required by pydot
                 if node.get_name() in ('graph', 'node', 'empty'):
                     continue
-                nodes[node.get_name()] = self.getNodeItemForNode(node, highlight_level)
+                nodes[node.get_name()] = self.getNodeItemForNode(node, highlight_level, scene=scene)
         for node in graph.nodes_iter():
             # hack required by pydot
             if node.get_name() in ('graph', 'node', 'empty'):
                 continue
-            nodes[node.get_name()] = self.getNodeItemForNode(node, highlight_level)
+            nodes[node.get_name()] = self.getNodeItemForNode(node, highlight_level, scene=scene)
+        return nodes
 
-        edges = {}
+    def parse_edges(self, graph, nodes, highlight_level, same_label_siblings, scene=None):
+        """Recursively searches all edges inside the graph and all subgraphs."""
+        # let pydot imitate pygraphviz api
+        graph.subgraphs_iter = graph.get_subgraph_list
+        graph.edges_iter = graph.get_edge_list
+
+        edges = {} # Empty dictionary
 
         for subgraph in graph.subgraphs_iter():
             subgraph.edges_iter = subgraph.get_edge_list
+            edges.update(self.parse_edges(subgraph, nodes, highlight_level,
+                same_label_siblings, scene=scene)[1]) # Get edges of subgraph
             for edge in subgraph.edges_iter():
                 self.addEdgeItem(edge, nodes, edges,
                                  highlight_level=highlight_level,
-                                 same_label_siblings=same_label_siblings)
+                                 same_label_siblings=same_label_siblings,
+                                 scene=scene)
 
         for edge in graph.edges_iter():
             self.addEdgeItem(edge, nodes, edges,
                              highlight_level=highlight_level,
-                             same_label_siblings=same_label_siblings)
+                             same_label_siblings=same_label_siblings,
+                             scene=scene)
 
-        return nodes, edges
+        return (nodes, edges)
```

Comment:
Why does the function return `nodes`? Where it is being called it only expects `edges` as the return value.

Reply:
Odd. On line #299 the function runs itself recursively, then gets the second element.

Looking at all the usages of this method, I'm not sure why this works. I explicitly remember testing the code for indigo successfully, I'll have to look at it again.